src/client/threads.js

```diff
@@ -318,13 +319,47 @@ NetsProcess.prototype.parseRPCResult = function (result) {
     return result;
 };
 
+function listToJSON(list) {
```

Comment:
I don't really like these as global functions - we should find somewhere better to declare them... It's not immediately clear where a better place would be though